modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -6333,6 +6375,16 @@ else if (msgLog.isDebugEnabled())
                             spi.failureDetectionTimeout() : spi.getSocketTimeout());
                     }
                 }
+                else if (msg instanceof TcpDiscoveryClientHeartbeatMessage) {
+                    TcpDiscoveryClientHeartbeatMessage hbmsg = (TcpDiscoveryClientHeartbeatMessage)msg;
+
+                    if (log.isDebugEnabled()) // TODO turn to debug
+                        log.debug("###  Received heartbeat message from node:" + hbmsg.creatorNodeId());
+
+                    this.metrics = hbmsg.metrics();
+
+                    this.lastHeartBeatTime=U.currentTimeMillis();
```

Comment:
Please use automatic format feature provided by IDEA (ALT + CTRL + L)
to configure codestyle please do:
idea -> setting-> code Style -> manger
and import file ${igniteProject}/idea/ignite_codeStyle.xml